lib/class-wp-theme-json-gutenberg.php

```diff
@@ -1106,85 +1132,57 @@ public function get_stylesheet( $type = 'all' ) {
 	/**
 	 * Merge new incoming data.
 	 *
-	 * @param WP_Theme_JSON_Gutenberg $incoming Data to merge.
-	 * @param string                  $update_or_remove Whether update or remove existing colors
-	 *                                                  for which the incoming data has a duplicated slug.
+	 * @param WP_Theme_JSON $incoming Data to merge.
 	 */
-	public function merge( $incoming, $update_or_remove = 'remove' ) {
-		$incoming_data = $incoming->get_raw_data();
-		$existing_data = $this->theme_json;
+	public function merge( $incoming, $origin ) {
+
+		$incoming_data    = $incoming->get_raw_data();
+		$this->theme_json = array_replace_recursive( $this->theme_json, $incoming_data );
 
 		// The array_replace_recursive algorithm merges at the leaf level.
 		// For leaf values that are arrays it will use the numeric indexes for replacement.
-		$this->theme_json = array_replace_recursive( $this->theme_json, $incoming_data );
+		// In those cases, what we want is to use the incoming value, if it exists.
+		//
+		// These are the cases that have array values at the leaf levels.
+		$properties   = array();
+
+		$properties[] = array( 'custom' );
+		$properties[] = array( 'spacing', 'units' );
 
-		// There are a few cases in which we want to merge things differently
-		// from what array_replace_recursive does.
-
-		// Some incoming properties should replace the existing.
-		$to_replace   = array();
-		$to_replace[] = array( 'custom' );
-		$to_replace[] = array( 'spacing', 'units' );
-		$to_replace[] = array( 'typography', 'fontSizes' );
-		$to_replace[] = array( 'typography', 'fontFamilies' );
-
-		// Some others should be appended to the existing.
-		// If the slug is the same than an existing element,
-		// the $update_or_remove param is used to decide
-		// what to do with the existing element:
-		// either remove it and append the incoming,
-		// or update it with the incoming.
-		$to_append   = array();
-		$to_append[] = array( 'color', 'duotone' );
```

Comment:
We're missing duotone from the list.

Naming: with the new behavior for merge `to_replace` / `to_append` no longer make sense, we could look for better names to both these. Perhaps `$properties` and `$propertis_per_origin`?

Reply:
I don't think duotone should be part of this list. For the duotone case, we don't have classes or CSS variables it is a server-side dynamic mechanism.

Reply:
Yeah, my point is that we're missing it: it's not in _any_ list at the moment (either `$properties` or `$to_append`).